arch/arm/core/aarch32/cortex_m/debug.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ARM Cortex-M debug monitor functions interface based on DWT
+ *
+ */
+
+#include <device.h>
+#include <aarch32/cortex_m/dwt.h>
+
+/**
+ * @brief Assess whether a debug monitor event should be treated as an error
+ *
+ * This routine checks the status of a debug monitor ()exception, and
+ * evaluates whether this needs to be considered as a processor error.
+ *
+ * @return true if the DM exception is a processor error, otherwise false
+ */
+bool z_arm_debug_monitor_event_error_check(void)
+{
+#if defined(CONFIG_CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_DETECTION_DWT)
+	/* Assess whether this debug exception was triggered
+	 * as a result of a null pointer (R/W) dereferencing.
+	 */
+	if (SCB->DFSR & SCB_DFSR_DWTTRAP_Msk) {
+		/* Debug event generated by the DWT unit */
+		if (DWT->FUNCTION0 & DWT_FUNCTION_MATCHED_Msk) {
+			/* DWT Comparator match used for */
+			printk("Null-pointer exception?\n");
+		}
+		__ASSERT((DWT->FUNCTION0 & DWT_FUNCTION_MATCHED_Msk) == 0,
+			"MATCHED flash should have cleared.");
+
+		return true;
+	}
+#endif /* CONFIG_CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_DETECTION_DWT */
+	return false;
```

Comment:
Might also want to treat `SCB_DFSR_BKPT_Msk` events as an error (since it would mean the system will be stuck in an infinite loop).

Reply:
that's probably fine, thanks
on the other hand, i wanted any external debug events to be recoverable, i.e. code not to get stuck here. That's why i wanted this PR to be transparent to everything else except for the comparator match...
i guess it's fine to add this flag for Breakpoints - but i am not 100% convinced.

Reply:
Update: I included your suggestion in the function , i think it makes sense.